navigation/tracking/scripts/pdaf.py

```diff
@@ -0,0 +1,250 @@
+
+Sub tasks: 
+
+    Use a kalam gain so that the filter is numerically stable. 
+    SINGULAR MATRICIES
+
+    TEST
+        - Visualize position estimates with clutter.
+        - Visualize velocity estimates with vectors. 
+        - Visualize validation gate. 
+        - Does the frequency of "zero detections within gate" correspond with p of detection? 
+
+    Observations: 
+        P pri and post are not symmetrical.
+        P post grows to values over 1000.
+"""
+
+
+class PDAF:
+    def __init__(self):
+        # x = [x, y, x', y']
+
+        self.time_step = 0.1
+        self.state_pri = np.ndarray(
+            (4,), buffer=np.array([0.0, 0.0, 0.0, 0.0]), dtype=float
+        )
+        self.P_pri = np.ndarray(
+            (4, 4),
+            buffer=np.array(
+                [
+                    [0.0, 0.0, 0.0, 0.0],
+                    [0.0, 0.0, 0.0, 0.0],
+                    [0.0, 0.0, 0.0, 0.0],
+                    [0.0, 0.0, 0.0, 0.0],
+                ]
+            ),
+            dtype=float,
+        )
+
+        self.state_post = np.ndarray((4,), dtype=float)
+        self.P_post = np.ndarray((4, 4), dtype=float)
+
+        self.L = np.ndarray((2, 2), dtype=float)
+
+        self.C = np.ndarray(
+            (2, 4),
+            buffer=np.array([[1.0, 0.0, 0.0, 0.0], [0.0, 1.0, 0.0, 0.0]]),
+            dtype=float,
+        )
+
+        self.A = np.ndarray(
+            (4, 4),
+            buffer=np.array(
+                [
+                    [1.0, 0.0, self.time_step, 0],
+                    [0, 1.0, 0, self.time_step],
+                    [0, 0, 1.0, 0],  # assuming constnat velocity
+                    [0, 0, 0, 1.0],  # assuming constnat velocity
+                ]
+            ),
+            dtype=float,
+        )
+
+        self.o_pri =  np.ndarray((2,), dtype=float) #predicted observation
+
+        self.Q = np.ndarray(
+            (4, 4),
+            buffer=np.array(
+                [[0.001, 0, 0, 0], [0, 0.001, 0, 0], [0, 0, 0.1, 0], [0, 0, 0, 0.1]]
+            ),
+            dtype=float,
+        )
+
+        self.R = np.ndarray((2, 2), buffer=np.array([[0.1, 0], [0, 0.1]]), dtype=float)
+
+        self.S = np.ndarray((2, 2), buffer=np.array([[0.1, 0], [0, 0.1]]), dtype=float)
+
+        self.validation_gate_scaling_param = (
+            2  # number of standard deviations we are willing to consider.
+        )
+
+        self.residual_vector = np.ndarray((2,), dtype=float)
+        self.p_no_match = 0.01  # probabiity that no observations matches the track
+        self.p_match_arr = np.ndarray(
+            (2,), dtype=float
+        )  # Lengt of this array will vary based on how many observations there are.
+        self.o_within_gate_arr = np.ndarray(
+            (2, 2), dtype=float
+        )  # Lengt of this array will vary based on how many observations there are.
+
+        self.n = 0  # used for N/M track manager
+        self.m = 0  # used for N/M track manager
+
```

Comment:
Leaking implementation from a track manager into the tracker itself should be avoided, since it reduces the modularity of tracker/manager.

If I haven't overlooked anything, it seems like these are part of stored state and actually not used internally in the PDAF, which means that it could be moved out without much trouble.

For the track manager, it might be a good idea to not use the PDAF object directly, and instead opt for something like 

```
@dataclass
class PDAF_2MN:
    tracker: PDAF
    m: int
    n: int
```